App/Sagas/TextileSagas.js

```diff
@@ -239,24 +239,52 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
-    for (const photo of photos.reverse()) {
+    const camera = yield select(TextileSelectors.camera)
+    const processed = camera && camera.processed ? camera.processed : []
+
+    let allPhotos = yield call(getAllPhotos)
+    if (camera === undefined || camera.processed === undefined || processed.length === 0) {
+      const ignoredPhotos = allPhotos.splice(1)
+      for (const uri of ignoredPhotos.map((photo) => photo.uri)) {
+        yield put(TextileActions.imageIgnore(uri))
+      }
+    }
+
+    let allProcessed = processed.reduce((o, item, index) => ({...o, [item]: { index }}), {})
```

Comment:
Nice! Storing an object containing the index as the value was just something needed for that example I pointed you to. The value could actually be anything. So it's fine as is, just an interesting point to mention...